knowledge-base/grid-show-confirm-cancel-dialog.md

```diff
@@ -0,0 +1,236 @@
+
+How to prompt the user to confirm canceling the edit/create before proceeding with other data operations(sort, filter, group etc.)?
+
+## Solution
+
+If the user tries to perform some other operation (such as sorting or filtering) while the Grid is in create or edit mode, the creating/editing will be canceled (check the last bullet of the [Editing Notes section]({%slug components/grid/editing/overview%}#notes) for more details). The `OnCancel` event will fire and you may handle it to display a Dialog to allow the user choose whether they want to cancel the editing or not.
+
+We provide predefined and custom dialogs and based on which one you want to proceed with, the implementation for achieving the desired scenario will be different:
+
+* [Predefined Confirm Dialog](#predefined-confirm-dialog) - this option is useful if you want to have a simple dialog with plain text a single confirmation button.
+
+* [Dialog Component](#dialog-component) - this solution allows full control over the dialog rendering. You may add any desired content there such as custom text, HTML elements or other components.
+
+This article lists samples for the Grid component. The suggested solutions, however, are applicable for the [TreeList]({%slug treelist-overview%}), too.
+
+### Predefined Confirm Dialog
+
+Use [Predefined Confirm Dialog]({%slug dialog-predefined%}#confirm) with the desired custom text. Additionally, you may get the details for the current item and add them to the text:
+
+* Handle the [`OnCancel`]({%slug grid-events%}#cud-events) event of the Grid
+* Display the Predefined Dialog in the `OnCancel` handler
+* Prevent the event or allow cancel edit/create depending on the user choice
+
+````CSHTML
+@*Only OnCancel event is handled for brevity*@
+
+@using System.ComponentModel.DataAnnotations @* for the validation attributes *@
+
+<TelerikGrid Data=@GridData
+             EditMode="@GridEditMode.Inline"
+             Pageable="true"
+             Sortable="true"
+             Height="500px"
+             OnCancel="@CancelHandler"
+             ShowColumnMenu="true">
+    <GridToolBar>
+        <GridCommandButton Command="Add" Icon="add">Add Employee</GridCommandButton>
+    </GridToolBar>
+    <GridColumns>
+        <GridColumn Field=@nameof(SampleData.ID) Title="ID" Editable="false" />
+        <GridColumn Field=@nameof(SampleData.Name) Title="Name" />
+        <GridCommandColumn>
+            <GridCommandButton Command="Save" Icon="save" ShowInEdit="true">Update</GridCommandButton>
+            <GridCommandButton Command="Edit" Icon="edit">Edit</GridCommandButton>
+            <GridCommandButton Command="Cancel" Icon="cancel" ShowInEdit="true">Cancel</GridCommandButton>
+        </GridCommandColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<SampleData> GridData { get; set; }
+
+    private bool cancelEdit { get; set; }
```

Comment:
Should be `CancelEdit` with a capital letter. Only local variables are camelCase.